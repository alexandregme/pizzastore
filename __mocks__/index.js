export const TOPPINGS = [{
  name:"topping 1",
  price:1,
  isChecked:true
},{
  name:"topping 2",
  price:2,
  isChecked:false
},{
  name:"topping 3",
  price:3,
  isChecked:false
},{
  name:"topping 4",
  price:4,
  isChecked:false
},{
  name:"topping 5",
  price:5,
  isChecked:false
},{
  name:"topping 6",
  price:6,
  isChecked:false
},{
  name:"topping 7",
  price:7,
  isChecked:false
},{
  name:"topping 8",
  price:8,
  isChecked:false
}];

export const DEFAULT_TOPPING = {
  name:"default topping",
  price:0,
  isChecked:false
};

export const CUSTOM_TOPPING = {
  name:"custom topping",
  price:1,
  isChecked:true
};

export const DEFAULT_PIZZA = {
  "name": "default pizza",
  "maxToppings": 0,
  "basePrice": 0,
  "totalPrice": 0,
  toppings:[]
};

export const CUSTOM_PIZZA = {
  "name": "custom pizza",
  "maxToppings": 1,
  "basePrice": 1,
  "totalPrice": 1,
  toppings:[{
    name:"topping 1",
    price:1,
    isChecked:true
  }]
};

export const PIZZAS = [{
  "name": "small",
  "maxToppings": 3,
  "basePrice": 9.89,
  toppings: TOPPINGS
},{
  "name": "medium",
  "maxToppings": 5,
  "basePrice": 10.89,
  toppings: TOPPINGS
}, {
  "name": "large",
  "maxToppings": null,
  "basePrice": 13.49,
  toppings: TOPPINGS
}];
