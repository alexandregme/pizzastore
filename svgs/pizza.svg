<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg  PUBLIC '-//W3C//DTD SVG 1.1//EN'  'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'>
<svg enable-background="new 0 0 58 58" version="1.1" viewBox="0 0 58 58" xml:space="preserve" xmlns="http://www.w3.org/2000/svg">
	<path d="m31.94 0.817c1.214 0.732 2.663 0.985 4.057 0.707 2.073-0.414 4.206 0.353 5.525 1.986 0.887 1.098 2.161 1.825 3.567 2.034 2.091 0.311 3.83 1.753 4.504 3.733 0.453 1.332 1.399 2.445 2.648 3.117 1.857 0.999 2.992 2.941 2.94 5.029-0.035 1.404 0.468 2.77 1.409 3.823 1.399 1.566 1.794 3.775 1.021 5.719-0.52 1.308-0.52 2.761 0 4.069 0.773 1.945 0.378 4.153-1.021 5.719-0.941 1.053-1.444 2.419-1.409 3.823 0.053 2.088-1.082 4.03-2.94 5.029-1.249 0.672-2.195 1.785-2.648 3.117-0.674 1.98-2.413 3.421-4.504 3.733-1.406 0.209-2.681 0.936-3.567 2.034-1.319 1.633-3.452 2.4-5.525 1.986-1.394-0.278-2.843-0.026-4.057 0.707-1.805 1.089-4.075 1.089-5.88 0-1.214-0.732-2.663-0.985-4.057-0.707-2.073 0.414-4.206-0.353-5.525-1.986-0.887-1.098-2.161-1.825-3.567-2.034-2.091-0.311-3.83-1.753-4.504-3.733-0.453-1.332-1.399-2.445-2.648-3.117-1.857-0.999-2.992-2.941-2.94-5.029 0.035-1.404-0.468-2.77-1.409-3.823-1.399-1.566-1.794-3.775-1.021-5.719 0.52-1.308 0.52-2.761 0-4.069-0.773-1.945-0.378-4.153 1.021-5.719 0.941-1.053 1.444-2.419 1.409-3.823-0.053-2.088 1.082-4.03 2.94-5.029 1.249-0.672 2.195-1.785 2.648-3.117 0.674-1.98 2.413-3.421 4.504-3.733 1.406-0.209 2.681-0.936 3.567-2.034 1.319-1.633 3.452-2.4 5.525-1.986 1.394 0.278 2.843 0.026 4.057-0.707 1.805-1.089 4.075-1.089 5.88 0z" fill="#FED070"/>
  <circle cx="29" cy="29" r="24" fill="#FAF8D2"/>
  <circle cx="21" cy="15" r="5" fill="#CD6E61"/>
  <circle cx="14" cy="32" r="5" fill="#CD6E61"/>
  <circle cx="39" cy="20" r="5" fill="#CD6E61"/>
  <circle cx="26" cy="44" r="5" fill="#CD6E61"/>
  <circle cx="41" cy="37" r="5" fill="#CD6E61"/>
  <circle cx="27.001" cy="27.999" r="5" fill="#CD6E61"/>
  <g fill="#D79C8C">
		<circle cx="20" cy="13" r="1"/>
    <circle cx="20" cy="17" r="1"/>
    <circle cx="23" cy="15" r="1"/>
    <circle cx="13" cy="30" r="1"/>
    <circle cx="13" cy="34" r="1"/>
    <circle cx="16" cy="32" r="1"/>
    <circle cx="38" cy="18" r="1"/>
    <circle cx="38" cy="22" r="1"/>
    <circle cx="41" cy="20" r="1"/>
    <circle cx="24.28" cy="42.572" r="1"/>
    <circle cx="25.889" cy="46.233" r="1"/>
    <circle cx="27.831" cy="43.195" r="1"/>
    <circle cx="42.09" cy="35.048" r="1"/>
    <circle cx="38.784" cy="37.299" r="1"/>
    <circle cx="42.126" cy="38.653" r="1"/>
    <circle cx="26.763" cy="25.775" r="1"/>
    <circle cx="25.366" cy="29.523" r="1"/>
    <circle cx="28.875" cy="28.697" r="1"/>
	</g>
  <path d="m29 17c-0.552 0-1-0.447-1-1s0.448-1 1-1c1.654 0 3-1.346 3-3 0-0.553 0.448-1 1-1s1 0.447 1 1c0 2.757-2.243 5-5 5z" fill="#88C057"/>
  <path d="m16.74 24.735c-2.095 0-4.051-1.326-4.749-3.419-0.175-0.524 0.108-1.091 0.632-1.265 0.525-0.175 1.09 0.108 1.265 0.632 0.522 1.569 2.227 2.421 3.795 1.897 0.525-0.171 1.09 0.109 1.265 0.634 0.174 0.523-0.109 1.09-0.633 1.265-0.522 0.174-1.052 0.256-1.575 0.256z" fill="#ED3F32"/>
  <path d="m37.739 45.999c-2.094 0-4.049-1.326-4.747-3.419-0.175-0.524 0.108-1.091 0.632-1.265 0.526-0.176 1.091 0.108 1.265 0.632 0.253 0.761 0.788 1.378 1.504 1.735 0.716 0.358 1.529 0.415 2.291 0.162 0.525-0.176 1.09 0.108 1.265 0.632s-0.108 1.091-0.632 1.265c-0.524 0.175-1.055 0.258-1.578 0.258z" fill="#88C057"/>
  <path d="m45.502 30.002c-0.843 0-1.677-0.216-2.434-0.639-0.482-0.27-0.655-0.879-0.385-1.361 0.27-0.481 0.877-0.654 1.361-0.385 0.701 0.393 1.511 0.487 2.281 0.27 0.771-0.219 1.411-0.725 1.802-1.425 0.269-0.481 0.877-0.655 1.361-0.385 0.482 0.27 0.655 0.879 0.385 1.361-0.652 1.166-1.719 2.009-3.003 2.372-0.452 0.128-0.911 0.192-1.368 0.192z" fill="#88C057"/>
  <path d="m29.502 38.002c-0.843 0-1.677-0.216-2.434-0.639-0.482-0.27-0.655-0.879-0.385-1.361 0.27-0.481 0.878-0.654 1.361-0.385 0.701 0.392 1.512 0.487 2.281 0.27 0.771-0.219 1.411-0.725 1.802-1.425 0.269-0.481 0.877-0.653 1.361-0.385 0.482 0.27 0.655 0.879 0.385 1.361-0.652 1.166-1.719 2.009-3.003 2.372-0.452 0.128-0.911 0.192-1.368 0.192z" fill="#ED3F32"/>
  <path d="m18.619 40.375c-0.396 0-0.77-0.235-0.927-0.625-0.208-0.512 0.04-1.095 0.551-1.302 1.533-0.621 2.275-2.374 1.655-3.907-0.208-0.512 0.04-1.095 0.551-1.302 0.512-0.207 1.095 0.039 1.302 0.552 1.035 2.555-0.202 5.476-2.757 6.511-0.123 0.05-0.25 0.073-0.375 0.073z" fill="#88C057"/>
  <path d="m35.789 33c-0.473 0-0.894-0.337-0.982-0.819-0.242-1.314 0.042-2.644 0.8-3.742 0.758-1.1 1.899-1.838 3.213-2.079 0.546-0.102 1.065 0.261 1.164 0.803 0.1 0.543-0.26 1.064-0.803 1.164-0.789 0.146-1.474 0.588-1.928 1.247-0.455 0.659-0.625 1.457-0.48 2.246 0.1 0.543-0.259 1.064-0.803 1.164-0.061 0.011-0.122 0.016-0.181 0.016z" fill="#ED3F32"/>
</svg>
