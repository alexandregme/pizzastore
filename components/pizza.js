import React from 'react';
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import { addToCart } from '../actions';
import Button from "./button";
import Toppings from "./toppings";
import { DEFAULT_PIZZA } from "../__mocks__";
import PizzaSVG from '../svgs/pizza.svg';

export const Pizza = ({pizza = DEFAULT_PIZZA, handleAddToCart = ()=>{}}) => (
  <React.Fragment>
    <div className="card bg-inverse text-white text-center mb-3">
        <PizzaSVG className="pizzaSVG" />
        <div className="card-body">
          <h5 id="pizzaName" className="card-title">Pizza size - {pizza.name}</h5>
          <p id="pizzaMaxToppings" className="card-text">Max Toppings - {pizza.maxToppings ? pizza.maxToppings : 'unlimited toppings!'}</p>
          <p id="pizzaBasePrice" className="card-text">Base price - ${pizza.basePrice}</p>
          <p id="pizzaTotalPrice" className="card-text">Total price - ${pizza.totalPrice}</p>
          <Toppings toppings={pizza.toppings} pizzaName={pizza.name}/>
          <Button
            label={'add to cart'}
            className="btn btn-warning text-gray-dark"
            handleClick={handleAddToCart}
          />
        </div>
    </div>
  </React.Fragment>
);

Pizza.propTypes = {
  pizza: PropTypes.object,
  handleAddToCart: PropTypes.func
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  handleAddToCart: () => dispatch(addToCart(ownProps.pizza))
});

export default connect(
  null,
  mapDispatchToProps
)(Pizza);
