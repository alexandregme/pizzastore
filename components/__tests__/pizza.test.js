import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Pizza } from '../pizza';
import Button from "../button";
import Toppings from "../toppings";
import { CUSTOM_PIZZA } from "../../__mocks__";

describe("Pizza", ()=>{
  let mountedPizza;
  beforeEach(()=>{
    mountedPizza = shallow(<Pizza />);
  });

  it('renders without crashing', () => {
    shallow(<Pizza />);
    expect(mountedPizza).toMatchSnapshot();
  });

  it('expect to have default value for pizza name', () => {
    let pizzaNameValue = mountedPizza.find('h5#pizzaName');
    expect(pizzaNameValue).toHaveLength(1);
    expect(pizzaNameValue.text()).toEqual('Pizza size - default pizza');
  });

  it('expect to have default value for pizza max toppings', () => {
    let pizzaMaxToppingValue = mountedPizza.find('p#pizzaMaxToppings');
    expect(pizzaMaxToppingValue).toHaveLength(1);
    expect(pizzaMaxToppingValue.text()).toEqual("Max Toppings - unlimited toppings!");
  });

  it('expect to have default value for base price', () => {
    let pizzaBasePriceValue = mountedPizza.find('p#pizzaBasePrice');
    expect(pizzaBasePriceValue).toHaveLength(1);
    expect(pizzaBasePriceValue.text()).toEqual("Base price - $0");
  });

  it('expect to have default value for total price', () => {
    let pizzaTotalPriceValue = mountedPizza.find('p#pizzaTotalPrice');
    expect(pizzaTotalPriceValue).toHaveLength(1);
    expect(pizzaTotalPriceValue.text()).toEqual("Total price - $0");
  });

  it('expect to have Toppings component', () => {
    expect(mountedPizza.find(Toppings)).toHaveLength(1);
  });

  it('expect to have Button component', () => {
    expect(mountedPizza.find(Button)).toHaveLength(1);
  });

  it('expect the return to be undefined to default function handleClick', () => {
    let button = mountedPizza.find(Button);
    expect(button.props().handleClick()).toBe(undefined);
  });
});

describe("Pizza Custom component props", ()=>{
  let mountedPizza;
  let props = {pizza:CUSTOM_PIZZA};

  beforeEach(()=>{
    mountedPizza = shallow(<Pizza {...props} />);
  });

  it('expect to have custom value for pizza name', () => {
    let pizzaNameValue = mountedPizza.find('h5#pizzaName');
    expect(pizzaNameValue).toHaveLength(1);
    expect(pizzaNameValue.text()).toEqual('Pizza size - custom pizza');
  });

  it('expect to have custom value for pizza max toppings', () => {
    let pizzaMaxToppingValue = mountedPizza.find('p#pizzaMaxToppings');
    expect(pizzaMaxToppingValue).toHaveLength(1);
    expect(pizzaMaxToppingValue.text()).toEqual("Max Toppings - 1");
  });

  it('expect to have custom value for base price', () => {
    let pizzaBasePriceValue = mountedPizza.find('p#pizzaBasePrice');
    expect(pizzaBasePriceValue).toHaveLength(1);
    expect(pizzaBasePriceValue.text()).toEqual("Base price - $1");
  });

  it('expect to have custom value for total price', () => {
    let pizzaTotalPriceValue = mountedPizza.find('p#pizzaTotalPrice');
    expect(pizzaTotalPriceValue).toHaveLength(1);
    expect(pizzaTotalPriceValue.text()).toEqual("Total price - $1");
  });
});
