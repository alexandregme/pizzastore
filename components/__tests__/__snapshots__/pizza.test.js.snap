// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pizza renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pizza />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="card bg-inverse text-white text-center"
      >
        <PizzaSVG
          className="pizzaSVG"
          viewBox="0 0 58 58"
          xmlns="http://www.w3.org/2000/svg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
            id="pizzaName"
          >
            default pizza
          </h5>
          <p
            className="card-text"
            id="pizzaMaxToppings"
          >
            0
          </p>
          <p
            className="card-text"
            id="pizzaBasePrice"
          >
            0
          </p>
          <Toppings
            pizzaName="default pizza"
            toppings={Array []}
          />
          <Button
            className="btn btn-warning text-gray-dark"
            handleClick={[Function]}
            label="add to cart"
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PizzaSVG
            className="pizzaSVG"
            viewBox="0 0 58 58"
            xmlns="http://www.w3.org/2000/svg"
          />,
          <div
            className="card-body"
          >
            <h5
              className="card-title"
              id="pizzaName"
            >
              default pizza
            </h5>
            <p
              className="card-text"
              id="pizzaMaxToppings"
            >
              0
            </p>
            <p
              className="card-text"
              id="pizzaBasePrice"
            >
              0
            </p>
            <Toppings
              pizzaName="default pizza"
              toppings={Array []}
            />
            <Button
              className="btn btn-warning text-gray-dark"
              handleClick={[Function]}
              label="add to cart"
            />
          </div>,
        ],
        "className": "card bg-inverse text-white text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "pizzaSVG",
            "viewBox": "0 0 58 58",
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-title"
                id="pizzaName"
              >
                default pizza
              </h5>,
              <p
                className="card-text"
                id="pizzaMaxToppings"
              >
                0
              </p>,
              <p
                className="card-text"
                id="pizzaBasePrice"
              >
                0
              </p>,
              <Toppings
                pizzaName="default pizza"
                toppings={Array []}
              />,
              <Button
                className="btn btn-warning text-gray-dark"
                handleClick={[Function]}
                label="add to cart"
              />,
            ],
            "className": "card-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "default pizza",
                "className": "card-title",
                "id": "pizzaName",
              },
              "ref": null,
              "rendered": "default pizza",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "card-text",
                "id": "pizzaMaxToppings",
              },
              "ref": null,
              "rendered": 0,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "card-text",
                "id": "pizzaBasePrice",
              },
              "ref": null,
              "rendered": 0,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "pizzaName": "default pizza",
                "toppings": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "btn btn-warning text-gray-dark",
                "handleClick": [Function],
                "label": "add to cart",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="card bg-inverse text-white text-center"
        >
          <PizzaSVG
            className="pizzaSVG"
            viewBox="0 0 58 58"
            xmlns="http://www.w3.org/2000/svg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
              id="pizzaName"
            >
              default pizza
            </h5>
            <p
              className="card-text"
              id="pizzaMaxToppings"
            >
              0
            </p>
            <p
              className="card-text"
              id="pizzaBasePrice"
            >
              0
            </p>
            <Toppings
              pizzaName="default pizza"
              toppings={Array []}
            />
            <Button
              className="btn btn-warning text-gray-dark"
              handleClick={[Function]}
              label="add to cart"
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PizzaSVG
              className="pizzaSVG"
              viewBox="0 0 58 58"
              xmlns="http://www.w3.org/2000/svg"
            />,
            <div
              className="card-body"
            >
              <h5
                className="card-title"
                id="pizzaName"
              >
                default pizza
              </h5>
              <p
                className="card-text"
                id="pizzaMaxToppings"
              >
                0
              </p>
              <p
                className="card-text"
                id="pizzaBasePrice"
              >
                0
              </p>
              <Toppings
                pizzaName="default pizza"
                toppings={Array []}
              />
              <Button
                className="btn btn-warning text-gray-dark"
                handleClick={[Function]}
                label="add to cart"
              />
            </div>,
          ],
          "className": "card bg-inverse text-white text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "pizzaSVG",
              "viewBox": "0 0 58 58",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="card-title"
                  id="pizzaName"
                >
                  default pizza
                </h5>,
                <p
                  className="card-text"
                  id="pizzaMaxToppings"
                >
                  0
                </p>,
                <p
                  className="card-text"
                  id="pizzaBasePrice"
                >
                  0
                </p>,
                <Toppings
                  pizzaName="default pizza"
                  toppings={Array []}
                />,
                <Button
                  className="btn btn-warning text-gray-dark"
                  handleClick={[Function]}
                  label="add to cart"
                />,
              ],
              "className": "card-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "default pizza",
                  "className": "card-title",
                  "id": "pizzaName",
                },
                "ref": null,
                "rendered": "default pizza",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "card-text",
                  "id": "pizzaMaxToppings",
                },
                "ref": null,
                "rendered": 0,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "card-text",
                  "id": "pizzaBasePrice",
                },
                "ref": null,
                "rendered": 0,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "pizzaName": "default pizza",
                  "toppings": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "btn btn-warning text-gray-dark",
                  "handleClick": [Function],
                  "label": "add to cart",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
